tests/mock_crust/churn.rs

```diff
@@ -107,11 +120,35 @@ fn random_churn<R: Rng>(rng: &mut R,
 
         None
     } else {
-        let proxy = rng.gen_range(0, len);
+        let mut proxy = rng.gen_range(0, len);
         let index = rng.gen_range(1, len + 1);
+
+        if nodes.len() > 16 {
+            let peer_1 = rng.gen_range(0, len);
+            let peer_2 = gen_range_except(rng, 0, len, Some(peer_1));
+            network.lost_connection(nodes[peer_1].handle.endpoint(),
+                                    nodes[peer_2].handle.endpoint());
+        }
+
         let config = Config::with_contacts(&[nodes[proxy].handle.endpoint()]);
 
         nodes.insert(index, TestNode::builder(network).config(config).create());
+
+        if nodes.len() > 16 {
+            if index <= proxy {
+                proxy += 1;
+            }
```

Comment:
new node may got inserted into a position before the proxy, hence push proxy_node's index by one.